File: Strings/Reverse String/JS/reverse_string.js
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function(s) {
  for (let i = 0, j = s.length - 1; i < j; i++, j--){
    let tmp = s[i];
    s[i] = s[j];
    s[j] = tmp;
  }
};

let s = ["h", "e", "l", "l", "o"];
reverseString(s);
console.log(s);

File: Easy/Strings/Reverse String/python/reverse_string.py
def reverseString(s):
    """
    Do not return anything, modify s in-place instead.
    """

    assert isinstance(s, list)
    s.reverse()
    return s


if __name__ == "__main__":
    test_s = ["h", "e", "l", "l", "o"]

    print(f"{test_s} translate to {reverseString(test_s)}")
